### ISSUE TITLE
Incorrect KEEP / UNDO gets fired under certain exit conditions

### ISSUE THREAD

**Issue by ugexe** [rakudo/rakudo#1214]
I expect `.print` (or .say, .note, etc), which returns `True`, to fire the KEEP phaser.
```
$ perl6 -e 'for 1..5 { KEEP print(" "); UNDO print("_");  .print; }'

1_2_3_4_5_
```
https://irclog.perlgeek.de/perl6-dev/2017-10-26#i_15359546

**Comment by skids** [comment]
Golf:
$ perl6 -e '{ KEEP print("K"); 42.Str; };'
$ perl6 -e '{ KEEP print("K"); 42; };'
K

Also this is actually being (wrongly) considered an unsuccessful block exit,
so it also affects hypothetical vars:

$ perl6 -e 'my $a = 0; { let $a = 1; KEEP print("K$_"); 42.Str; }; $a.say;' # Should print "K421"
0
$ perl6 -e 'my $a = 0; { let $a = 1; KEEP print("K$_"); 42; }; $a.say;' # Should print "K421"
K421

"sink 42" also kills it, but probably should since the block then returns Nil which is truly unsuccessful

ISTR code to short-circuit "useless use" messages when a method form is seen.  That's where to start looking IMO.

**Comment by skids** [comment]
The above commit was spectested when merged with HEAD and note there is a previous commit in that branch for another issue.  The branch itself was kept back to avoid merge noise.  I don't know if that is precisely the correct flavor of wanting to use there... that whole system confuses me a bit.

**Comment by lizmat** [comment]
This appears to be fixed since 2021.10 release, so closing
